### DISCUSSION TITLE
What is the recommended way to capture per-route latency/error metrics?

### DISCUSSION THREAD

**Discussion by petro2050** [aws/aws-lambda-dotnet#1910] [Q&A] [answered]
For an ASP.NET Web API app running as a lambda, how should we go about capturing per-route metrics? Lambda monitoring tab shows metrics at lambda level which are not that useful. I can think of enabling route metrics at API Gateway level, but I like the convenience of using a greedy proxy route.

What does the team recommend?

**Comment by ashishdhingra** [comment, marked as answer]
@petro2050 Good afternoon. Please advise if you tried using [Enhancing Observability in the AWS SDK for .NET with OpenTelemetry](https://aws.amazon.com/blogs/developer/enhancing-observability-in-the-aws-sdk-for-net-with-opentelemetry/). Also refer [Announcing the general availability of AWS .NET OpenTelemetry libraries](https://aws.amazon.com/blogs/dotnet/announcing-the-general-availability-of-aws-net-opentelemetry-libraries/).
